libbindgen/src/ir/context.rs

```diff
@@ -1255,8 +1255,9 @@ impl<'ctx, 'gen> Iterator for AssertNoDanglingItemIter<'ctx, 'gen>
         }
 
         for sub_id in sub_types {
-            if let Some(value) = self.seen.insert(id, sub_id) {
-                self.to_iterate.push_back(value);
+            // We've never visited this sub item before.
+            if self.seen.insert(sub_id, id).is_none() {
+                self.to_iterate.push_back(sub_id);
```

Comment:
Yes, this looks better.

Nitpickiest of nitpicks: I'd prefer if the comment was *inside* the if's consequent.

/me ducks

Thanks! And sorry for pushing back on this, but I think it is the right thing to do